### ISSUE TITLE
Create Conv layer

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#4]
## Task
* Create a layer child of BaseLayer
* Create 2 versions the naive and with im2col
* Add proper constructor on LayerContainer
* Create tests on test/layers, use data from cs231n exercises

## MNIST Test
Create the following model based on [ThreeLayerNet](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/cs231n/classifiers/cnn.py)
conv --> Relu --> Maxpool --> FC --> Relu --> FC --> Softmax
Use the multi-class cross entropy and train on MNIST 

## References
* [Book AI](https://leonardoaraujosantos.gitbooks.io/artificial-inteligence/content/convolution_layer.html)
* [Notes on Backprop](http://www.slideshare.net/kuwajima/cnnbp)
* [Video Hugo](https://www.youtube.com/watch?v=cDdpwAIsuD8&index=74&list=PL6Xpj9I5qXYEcOhn7TqghAJ6NAPrNmUBH)
* [Cs231n Notes](http://cs231n.github.io/convolutional-networks/)
* [Notebook with tensorflow on MNIST](https://github.com/jeffschecter/sandbox/blob/master/MNIST_convnet.ipynb)
* [Layers from cs231n](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/cs231n/layers.py)
* [im2col cs231n](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment3/cs231n/im2col.py)
* [Faster version cs231n](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment3/cs231n/fast_layers.py)
* [Blog on backpropagation with im2col](https://sunshineafternoonweb.wordpress.com/)
* [Cs231n Cython implementation im2col/col2im](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment3/cs231n/im2col_cython.pyx)
* [Chainer Convolution source](https://github.com/pfnet/chainer/blob/master/chainer/functions/connection/convolution_2d.py)
* [Matlab sum multiple dimensions](https://uk.mathworks.com/matlabcentral/newsreader/view_thread/279051)
